### DISCUSSION TITLE
Isar default path

### DISCUSSION THREAD

**Discussion by Spyy004** [isar/isar#1220] [Help]
So everything was fine till yesterday but today when I ran my code, it showed an error stating Isar.open() needs a directory path which is a required param. I never use to pass the directory to the Isar.open() function but since now it is asking for it, now I need to know where all the previous data is stored because I don't pass the exact path, I lose all my data. Help me out on how i can solve this issue please. I am using v 3.0.5

**Comment by simc** [comment]
You accidentally upgraded to 3.1.0

**Reply by Spyy004** [reply]
Yeah exactly!
So in future when 3.1.0 becomes mandatory. How do i make sure that things don't fail after migrating?

**Comment by Dosenbiiir** [comment]
We have the same problem with the new `open()` implementation. We need to access the old default location, so our users don't loose their data if we upgrade to the new Isar version.

The documentation for 3.1.0 'suggests' `getApplicationDocumentsDirectory()` which leads to ".../com.packagename/app_flutter/". At least for Android 13.

On Android 13 the old location where Isar puts the db file is ".../com.packagename/files/". I was not sure if the behaviour differs on other Android versions or other platforms like iOS. Thats why I checked the source code from version 3.0.5 and  it seems that Isar is written in Rust. I don't know Rust but I gave my best. 
The default path used in Rust code is set in different rust files depending on the used platform. 
For Android the path is set from Java via JNI: `context.getFilesDir().getAbsolutePath()`.

For iOS and MacOS it calls `NSSearchPathForDirectoriesInDomains(...)`from Rust but this is where i got stuck as I don't know iOS.

Other platforms uses: `Some(dirs::config_dir()?.to_str()?.to_string())` where i also got stuck.

It would be great if anyone can help out :)
